arkouda/suffix_array.py

```diff
@@ -0,0 +1,469 @@
+    by other arkouda functions. It is very similar to Strings and the difference is
+    that its content is int arrays instead of strings.
+
+    Attributes
+    ----------
+    offsets : pdarray
+        The starting indices for each suffix array
+    bytes : pdarray
+        The raw integer indices of all suffix arrays
+    size : int
+        The number of suffix arrays in the array
+    nbytes : int
+        The total number of indices in all suffix arrays
+        We have the same number indices as the number of characters/suffixes in strings
+    ndim : int
+        The rank of the array (currently only rank 1 arrays supported)
+    shape : tuple
+        The sizes of each dimension of the array
+    dtype : dtype
+        The dtype is np.int
+    logger : ArkoudaLogger
+        Used for all logging operations
+
+    Notes
+    -----
+    SArrays is composed of two pdarrays: (1) offsets, which contains the
+    starting indices for each string's suffix array  and (2) bytes, which contains the
+    indices of all suffix arrays, no any spliter between two index arrays.
+    """
+
+    BinOps = frozenset(["==", "!="])
+    objtype = "int"
+
+    def __init__(
+        self,
+        offset_attrib: Union[pdarray, np.ndarray],
+        bytes_attrib: Union[pdarray, np.ndarray],
+    ) -> None:
+        """
+        Initializes the SArrays instance by setting all instance
+        attributes, some of which are derived from the array parameters.
+
+        Parameters
+        ----------
+        offset_attrib : Union[pdarray, np.ndarray,array]
+            the array containing the offsets
+        bytes_attrib : Union[pdarray, np.ndarray,array]
+            the array containing the suffix array indices
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        RuntimeError
+            Raised if there's an error converting a Numpy array or standard
+            Python array to either the offset_attrib or bytes_attrib
+        ValueError
+            Raised if there's an error in generating instance attributes
+            from either the offset_attrib or bytes_attrib parameter
+        """
+        if isinstance(offset_attrib, pdarray):
+            self.offsets = offset_attrib
+        else:
+            try:
+                self.offsets = create_pdarray(offset_attrib)
+            except Exception as e:
+                raise RuntimeError(e)
+        if isinstance(bytes_attrib, pdarray):
+            self.bytes = bytes_attrib
+        else:
+            try:
+                self.bytes = create_pdarray(bytes_attrib)
+            except Exception as e:
+                raise RuntimeError(e)
+        try:
+            self.size = self.offsets.size
+            self.nbytes = self.bytes.size
+            self.ndim = self.offsets.ndim
+            self.shape = self.offsets.shape
+        except Exception as e:
+            raise ValueError(e)
+        self.dtype = int64
```

Comment:
Note, if we want to keep this in line with `Strings` then we should add
`self.name: Optional[str] = None`  here along with updates to the register/attach, etc. methods.

Note, the attach/register suite of methods now use a form of `user_defined_name + ".offsets/bytes"` etc.  where we now use a dot character instead of an underscore.

It doesn't necessarily need to be changed now but it should be immediate follow-on work to be done ASAP.

Reply:
👍